Add support for ONNXRuntime 1.17.1

ktf opened the pull request:
Add support for ONNXRuntime 1.17.1

Notice that this includes a fix to give precedence to the old ONNXRuntime.

Without this, it will currently prefer (non-working) local installations.

ktf committed:
Use C++14 namespaces

alibuild commented:
Error while checking build/O2/fullCI for 98a87bfe926d21634cf537ce21b014c671f93b16 at 2024-04-11 14:06:
```
## sw/BUILD/O2-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
/sw/SOURCES/O2/13012-slc8_x86-64/0/Detectors/ZDC/fastsimulation/include/FastSimulations.h:23:10: fatal error: onnxruntime_cxx_api.h: No such file or directory
ninja: build stopped: subcommand failed.
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/13012/98a87bfe926d21634cf537ce21b014c671f93b16/build_O2_fullCI/pretty.html).

ktf committed:
Add support for ONNXRuntime 1.17.1

Notice that this includes a fix to give precedence to the old ONNXRuntime.

Without this, it will currently prefer (non-working) local installations.
